cap-async-std/src/fs/dir.rs

```diff
@@ -1,19 +1,27 @@
-#[cfg(unix)]
-use crate::os::unix::net::{UnixDatagram, UnixListener, UnixStream};
-use crate::{
-    fs::{DirBuilder, File, Metadata, OpenOptions, Permissions, ReadDir},
-    sys,
-};
-#[cfg(unix)]
-use async_std::os::unix::io::{AsRawFd, FromRawFd, IntoRawFd, RawFd};
-#[cfg(windows)]
-use async_std::os::windows::io::{AsRawHandle, FromRawHandle, IntoRawHandle, RawHandle};
+use crate::fs::{DirBuilder, File, Metadata, OpenOptions, Permissions, ReadDir};
 use async_std::{fs, io};
 use std::{
     fmt,
+    mem::ManuallyDrop,
     path::{Path, PathBuf},
 };
 
+cfg_if::cfg_if! {
+    if #[cfg(any(unix, target_os = "fuchsia"))] {
+        use crate::os::unix::net::{UnixDatagram, UnixListener, UnixStream};
+        use cap_primitives::fs::{canonicalize, link, mkdir, open, stat, symlink, unlink, FollowSymlinks};
+        use async_std::os::unix::io::{AsRawFd, FromRawFd, IntoRawFd, RawFd};
+    } else if #[cfg(windows)] {
+        use cap_primitives::fs::{symlink_dir, symlink_file};
+        use async_std::os::windows::io::{AsRawHandle, FromRawHandle, IntoRawHandle, RawHandle};
+    } else if #[cfg(target_os = "wasi")] {
+        use async_std::os::wasi::{
+            fs::OpenOptionsExt,
+            io::{AsRawFd, IntoRawFd},
+        };
+    }
+}
```

Comment:
I think we can avoid `cfg_if` here and in the other places that do this, because the conditions are already disjoint, so we don't need `else`. To avoid repeating the condition for multiple use statements we can merge them:
```rust
use {
   crate::os::unix::net::UnixDataGram,
   cap_primitives::fs::canonicalize,
}
so that one `cfg` shouldsuffiice.

Reply:
Oh niiice! I love this! Lemme give it a whirl!